Feature/system update (#46)
Linuxfabrik/lfops: roles/system_update/README.md

# Ansible Role system_update

> This role configures automatic system updates.

Reviewer: Please add some more details, what would you want to know before you would choose to use this role? how often are the updates? explain the notify-mail, etc...

FQCN: linuxfabrik.lfops.system_update

Tested on

* Debian
* RHEL 7 (and compatible)
* RHEL 8 (and compatible)
* Fedora

## Requirements

### Roles:

* at
* mailx

## Tags

| Tag   				| What it does                               		 	  |
| ---   				| ------------                              		 	  |
| system_uodate 		| Sets up automatic system update via systemd timer.	  |
| system_update__state 	| Determines whether notify-and-schedule.timer is enabled |

## Role Variables

### Mandatory

#### system_update__notification_recipients

> The recipients of update and reboot notifications. This can consist of multiple addresses separated by whitespaces.

Reviewer: only update and reboot notifications, or also the notify-and-schedule ones?

The hour when to check for updates and to notify about them.

Example:
```yaml
system_update__notify_time_hour: '09'
```

#### system_update__notify_time_minute

The minute when to check for updates and to notify about them.

Example:
```yaml
system_update__notify_time_minute: '30'
```

#### system_update__update_time

The time when to actually execute the updates and to reboot.

Example:
```yaml
system_update__update_time: '04:04 + 1 days'
```

### Optional

#### system_update__update_enabled

Default:
```yaml
system_update__update_enabled: true
```

#### system_update__update_on_calendar

Allow setting via legacy variables, but it also allows setting update_on_calendar directly.

Default:
```yaml
system_update__update_on_calendar: '{{ system_update__notify_time_wday }} {{ system_update__notify_time_hour }}:{{ system_update__notify_time_minute }}'
```

Example:
```yaml
system_update__update_on_calendar: 'mon 09:30'
```

#### icinga2_api_user

The Icinga2 API User used to set downtimes.

Default:
```yaml
icinga2_api_user: ''
```

#### icinga2_api_password

The password for the Icinga2 API User used to set downtimes.

Default:
```yaml
icinga2_api_password: ''
```

> #### icinga2_master

Reviewer: same as above

## Requirements

### Roles

* linuxfabrik.lfops.at
* linuxfabrik.lfops.mailx

## Tags

| Tag   				| What it does                               		 	  |
| ---   				| ------------                              		 	  |
| system_uodate 		| Sets up automatic system update via systemd timer.	  |

> | system_update:state 	| Determines whether notify-and-schedule.timer is enabled |

Reviewer: use spaces instead of tabs

#### system_update__notification_recipients

> A list of recipients to notify about update schedules, the updates themselves and reboots.

Reviewer: ```suggestion
A list of email recipients to notify about update schedules, the updates themselves and reboots.
```

Example:
```yaml
system_update__notification_recipients:

> 	- 'info@example.com'

Reviewer: use two spaces instead of tabs

#### system_update__new_configfile_notification_recipients

> A list of recipients to notify if new versions config files have been found.

Reviewer: ```suggestion
A list of email recipients to notify if new versions config files have been found.
```

```

#### system_update__update_on_calendar
#### system_update__icinga2_api_user

Allow setting via legacy variables, but it also allows setting update_on_calendar directly.
The Icinga2 API User used to set the downtime for the corresponding host and all it's services.

Default:
Example:
```yaml

> system_update__update_on_calendar: '{{ system_update__notify_time_wday }} {{ system_update__notify_time_hour }}:{{ system_update__notify_time_minute }}'

Reviewer: wrong way around - keep `system_update__update_on_calendar`, remove `system_update__notify_time_wday }} {{ system_update__notify_time_hour }}:{{ system_update__notify_time_minute`

#### system_update__icinga2_api_password

The password for the Icinga2 API User used to set the downtime for the corresponding host and all it's services.

Example:
```yaml
system_update__icinga2_api_password: '1234abcd'
```

#### system_update__icinga2_master

The hostname or ip address of the Icinga2 master instance where to set the downtime for the corresponding host and all it's services.

Example:
```yaml
system_update__icinga2_master: 'icinga.example.com'
```

#### system_update__rocketchat_url

The URL to a potential Rocket.Chat Server to send notifications about the updates to.

Example:
```yaml
system_update__rocketchat_url: 'rocketchat.example.com'
```

#### system_update__rocketchat_msg_suffix

A suffix to the Rocket.Chat notifications. This can be used to mention other users.

Example:
```yaml
system_update__rocketchat_msg_suffix: '@administrator'
```

#### system_update__post_update_code

> This codeblock will be executed after the updates have been installed and before a potentially scheduled reboot.

Reviewer: ```suggestion
This codeblock will be executed after the updates have been installed and before a potential reboot.
```
